crates/node/src/service.rs

```diff
@@ -527,6 +531,15 @@ pub fn new_full(
         );
     }
 
+    if let Some(ref l1_messages_config) = l1_messages {
+        let config = L1MessagesWorkerConfig::new_from_file(l1_messages_config)
+            .map_err(|e| sc_service::error::Error::Other(e.to_string()))?;
+        task_manager.spawn_essential_handle().spawn(
```

Comment:
`spawn_essential_handle` make it so that if this thread returns or panic, the whole node is shut down.
Imo this should not be the case. If we allow nodes to run without l1-worker we can allow the  service to go down without shuting down everything

Reply:
There is another method for non-essential services

Reply:
Followed instructions from https://github.com/keep-starknet-strange/madara/issues/1177 but will refactor into `spawn_handle`